Bottom: fccaacf1da4e959822b781f12cdfa516385cacc6
Top:    1ef7c1684425775219336a1f60f500b5aa0abe51
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 17:22:51 +0000

main.c: Initialise crosshair unconditionally

Since we no longer generate graphics contexts from InitCrosshair(),
there is no need to check gui->gui before calling it.


---

diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index 27bcc71..252e2bf 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -330,7 +330,7 @@ hid_batch_init ()
 {
   memset (&batch_hid, 0, sizeof (HID));
   memset (&batch_graphics, 0, sizeof (HID_DRAW));
-  memset (&batch_graphics_ckass, 0, sizeof (HID_DRAW_CLASS));
+  memset (&batch_graphics_class, 0, sizeof (HID_DRAW_CLASS));
 
   common_nogui_init (&batch_hid);
 
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 00c439c..e2626b6 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -4131,7 +4131,7 @@ hid_lesstif_init ()
   lesstif_graphics_class.fill_polygon   = lesstif_fill_polygon;
   lesstif_graphics_class.fill_rect      = lesstif_fill_rect;
 
-  lesstif_graphics.draw_pcb_polygon    = common_gui_draw_pcb_polygon;
+  lesstif_graphics_class.draw_pcb_polygon = common_gui_draw_pcb_polygon;
 
   lesstif_graphics.klass = &lesstif_graphics_class;
   common_draw_helpers_init (&lesstif_graphics);
diff --git a/src/main.c b/src/main.c
index 04a116c..58ba024 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1990,8 +1990,7 @@ main (int argc, char *argv[])
 
   ResetStackAndVisibility ();
 
-  if (gui->gui)
-    InitCrosshair ();
+  InitCrosshair ();
   InitHandler ();
   InitBuffers ();
   SetMode (ARROW_MODE);
